geonode/base/forms.py

```diff
@@ -594,7 +594,7 @@ def __init__(self, *args, **kwargs):
         self.fields["layers"].label_from_instance = self.label_from_instance
 
     layers = MultipleChoiceField(
-        choices=get_user_choices(),
+        choices=get_user_choices,
```

Comment:
By doing this, we let Django update the choices list every time the page is loaded. In this way we always have all the datasets available